libraries/chain/db_management.cpp

```diff
@@ -87,12 +96,23 @@ void database::reindex(fc::path data_dir, const genesis_state_type& initial_allo
          wlog( "Dropped ${n} blocks from after the gap", ("n", dropped_count) );
          break;
       }
-      apply_block(*block, skip_witness_signature |
-                          skip_transaction_signatures |
-                          skip_transaction_dupe_check |
-                          skip_tapos_check |
-                          skip_witness_schedule_check |
-                          skip_authority_check);
+      if( i < undo_point )
+         apply_block(*block, skip_witness_signature |
+                             skip_transaction_signatures |
+                             skip_transaction_dupe_check |
+                             skip_tapos_check |
+                             skip_witness_schedule_check |
+                             skip_authority_check);
+      else
+      {
+         _undo_db.enable();
+         push_block(*block, skip_witness_signature |
+                            skip_transaction_signatures |
+                            skip_transaction_dupe_check |
+                            skip_tapos_check |
+                            skip_witness_schedule_check |
+                            skip_authority_check);
+      }
    }
    _undo_db.enable();
```

Comment:
This line need to be removed? We may have enabled it earlier. Also possible it hasn't been disabled.